### PR TITLE
refactor(workspace): Image can be queried using dataSource

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:
Images can be queried using dataSource.
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
Ulanqab1
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDesktop_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDesktop_basic -timeout 360m -parallel 4
=== RUN   TestAccDesktop_basic
=== PAUSE TestAccDesktop_basic
=== CONT  TestAccDesktop_basic
--- PASS: TestAccDesktop_basic (1660.90s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace    1660.943s

beijing4
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDesktop_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDesktop_basic -timeout 360m -parallel 4
=== RUN   TestAccDesktop_basic
=== PAUSE TestAccDesktop_basic
=== CONT  TestAccDesktop_basic
--- PASS: TestAccDesktop_basic (1745.76s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace       1745.828s

```

### CODE REVIEW

File: huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_desktop_test.go [hunk 1]
```diff
@@ -130,14 +130,19 @@ func testAccDesktop_basic_step1(rName string) string {
...
 %[1]s
 
+data "huaweicloud_images_images" "test" {
+  name       = "COPY_code_WORKSPACE_WIN_2019_EN_20230116"
+  visibility = "market"
+}
+
 locals {
   data_volume_sizes = [50, 70]
 }
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_desktop_test.go]:
```suggestion
data "huaweicloud_images_images" "test" {
  visibility = "market"
}

locals {
  data_volume_sizes = [50, 70]
  image_ids         = [for o in data.huaweicloud_images_images.test.images : o.id if strcontains(o.name, "WORKSPACE")]
}
```
using this code to obtain the available image IDs
```
resource "huaweicloud_workspace_desktop" "test" {
  ...
  image_id = try(image_ids[0], "")
}
And using this code to create the desktop using queried image ID.

Comment (wuzhuanhong) [reply huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_desktop_test.go]: fixed

**Comment by wuzhuanhong** [comment]
region="cn-east-3"
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDesktop_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDesktop_basic -timeout 360m -parallel 4
=== RUN   TestAccDesktop_basic
=== PAUSE TestAccDesktop_basic
=== CONT  TestAccDesktop_basic
--- PASS: TestAccDesktop_basic (1382.11s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 1382.196s

region="cn-north-4"
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDesktop_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDesktop_basic -timeout 360m -parallel 4
=== RUN   TestAccDesktop_basic
=== PAUSE TestAccDesktop_basic
=== CONT  TestAccDesktop_basic
--- PASS: TestAccDesktop_basic (1251.01s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 1251.096s

File: huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_desktop_test.go [hunk 2]
```diff
@@ -239,7 +244,7 @@ locals {
 resource "huaweicloud_workspace_desktop" "test" {
   flavor_id         = "workspace.x86.ultimate.large4"
   image_type        = "market"
-  image_id          = "4d698843-d653-4ffd-80be-4f47a0cabce0"
+  image_id          = try(data.huaweicloud_images_images.test.images[0].id, "")
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_desktop_test.go]:
```suggestion
  image_id          = try(data.huaweicloud_images_images.test.images[1].id, "")
```

Comment (wuzhuanhong) [reply huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_desktop_test.go]: fixed

**Comment by wuzhuanhong** [comment]
region="cn-south-1"
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDesktop_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDesktop_basic -timeout 360m -parallel 4
=== RUN   TestAccDesktop_basic
=== PAUSE TestAccDesktop_basic
=== CONT  TestAccDesktop_basic
--- PASS: TestAccDesktop_basic (1904.87s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 1904.916s


region="cn-north-4"
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDesktop_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDesktop_basic -timeout 360m -parallel 4
=== RUN   TestAccDesktop_basic
=== PAUSE TestAccDesktop_basic
=== CONT  TestAccDesktop_basic
--- PASS: TestAccDesktop_basic (1559.48s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 1559.607s

**Pull request merged**